src/rgw/CMakeLists.txt

```diff
@@ -249,38 +249,39 @@ endif()
 
 target_link_libraries(rgw_common
   PRIVATE
-    global
-    cls_2pc_queue_client
-    cls_cmpomap_client
-    cls_lock_client
-    cls_log_client
-    cls_otp_client
-    cls_refcount_client
-    cls_rgw_client
-    cls_rgw_gc_client
-    cls_timeindex_client
-    cls_user_client
-    cls_version_client
-    librados
-    rt
-    ICU::uc
-    OATH::OATH
-    dmclock::dmclock
-    ${CURL_LIBRARIES}
-    ${EXPAT_LIBRARIES}
-    ${ARROW_LIBRARIES}
-    ${ARROW_FLIGHT_LIBRARIES}
-    ${ALLOC_LIBS}
+  global
+  cls_2pc_queue_client
+  cls_cmpomap_client
+  cls_lock_client
+  cls_log_client
+  cls_otp_client
+  cls_refcount_client
+  cls_rgw_client
+  cls_rgw_gc_client
+  cls_timeindex_client
+  cls_user_client
+  cls_version_client
+  librados
+  $<$<AND:$<BOOL:UNIX>,$<NOT:$<BOOL:APPLE>>>:rt>
```

Comment:
yikes. i guess this syntax makes the linkage to `rt` conditional?

could this be a separate `target_link_libraries()` call below, guarded by a corresponding `if()` block? it looks like you used this elsewhere:
```cmake
if(NOT WIN32 AND NOT APPLE)
```
and please undo the indentation changes here

Reply:
Ack.

yikes it is! however, I had some fun figuring CMake generators. I got inspired by jaegertracing 
```
src/jaegertracing/opentelemetry-cpp/CMakeLists.txt:
  231:         "MultiThreaded$<$<CONFIG:Debug>:Debug>$<$<STREQUAL:${VCPKG_CRT_LINKAGE},dynamic>:DLL>"

src/jaegertracing/opentelemetry-cpp/third_party/ms-gsl/CMakeLists.txt:
  48:     $<$<CXX_COMPILER_ID:MSVC>:

src/jaegertracing/opentelemetry-cpp/third_party/ms-gsl/tests/CMakeLists.txt:
   53:         $<$<CXX_COMPILER_ID:MSVC>:
   61:           $<$<VERSION_LESS:$<CXX_COMPILER_VERSION>,1920>: # VS2015
   63:             $<$<NOT:$<CONFIG:Debug>>: # Release, RelWithDebInfo
   68:         $<$<CXX_COMPILER_ID:Clang>:
   81:           $<$<EQUAL:${GSL_CXX_STANDARD},14>: # no support for [[maybe_unused]]
  100:         $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>>:
  112:           $<$<EQUAL:${GSL_CXX_STANDARD},14>: # no support for [[maybe_unused]]
  117:         $<$<CXX_COMPILER_ID:Clang>:
  118:           $<$<AND:$<VERSION_GREATER:$<CXX_COMPILER_VERSION>,4.99>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,6>>:
```

I'm reverting this and adding an `if()`